tasks/selinux.yml

```diff
@@ -0,0 +1,57 @@
+# SPDX-License-Identifier: MIT
+---
+- block:
+    - name: Set the fence-virt/fence-agents port to _ha_cluster_selinux
+      set_fact:
+        _ha_cluster_selinux:
+          - {'ports': '1229', 'proto': 'tcp', 'setype': 'cluster_port_t',
+             'state': '{{ _state_value }}', 'local': 'true'}
+      when:
+        - (
+            'fence-virt' in ha_cluster_fence_agent_packages
+            or
+            'fence-virt' in ha_cluster_extra_packages
+            or
+            'fence-agents-all' in ha_cluster_fence_agent_packages
+            or
+            'fence-agents-all' in ha_cluster_extra_packages
+          )
+
+    - name: Get the high-availability tcp service ports
```

Comment:
you can use `firewall-cmd --service=high-availability --get-ports` to get the ports - here is the output:
```
>sudo firewall-cmd --info-service=high-availability
high-availability
  ports: 2224/tcp 3121/tcp 5403/tcp 5404/udp 5405-5412/udp 9929/tcp 9929/udp 21064/tcp
  protocols: 
  source-ports: 
  modules: 
  destination: 
  includes: 
  helpers: 
```

Reply:
In fact, you can eliminate using `set_fact` altogether:
```yaml
    - name: get firewall ports
      command: firewall-cmd --info-service=high-availability
      register: __out
    - name: construct selinux args
      debug:
        msg: selinux_ports {{ selinux_ports | to_nice_json }}
      vars:
        _state_value: enabled
        __pat: '(?sm)^.* ports: ([^\n]+)\n.*$'
        __port_list: "{{ __out.stdout | regex_replace(__pat, '\\1') | split |
          map('split', '/') }}"
        selinux_ports: |
          {% set selinux_ports = [] %}
          {% for port_proto in __port_list %}
          {%   set item = {'ports': port_proto[0], 'proto': port_proto[1],
                           'setype': 'cluster_port_t',
                           'state': _state_value, 'local': true} %}
          {%   set _ = selinux_ports.append(item) %}
          {% endfor %}
          {{ selinux_ports }}
```